File: CS/Reporting-How-To-Use-Object-Data-Source/MVC/CS/Reporting_ObjectDS_Mvc/Controllers/HomeController.cs
using DevExpress.DataAccess.ObjectBinding;
using System.Collections.Generic;
using System.Web.Mvc;

namespace Reporting_ObjectDS_Mvc.Controllers
{
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Designer() {
            Models.ReportDesignerModel model = new Models.ReportDesignerModel();
            ObjectDataSource dataSource = new ObjectDataSource();
            dataSource.Name = "EmployeeObjectDS";
            dataSource.DataSource = typeof(Employees.EmployeeList);
            dataSource.Constructor = ObjectConstructorInfo.Default;
            dataSource.DataMember = "Items";
            dataSource.RebuildResultSchema();
            model.DataSources = new Dictionary<string, object>();
            model.DataSources.Add("EmployeeObjectDS", dataSource);
            return View(model);
        }

        public ActionResult Viewer() {
            return View();
        }
    }
}

File: CS/Reporting-How-To-Use-Object-Data-Source/MVC/VB/Reporting_ObjectDS_Mvc/Controllers/HomeController.vb
Imports DevExpress.DataAccess.ObjectBinding
Imports System.Collections.Generic
Imports System.Web.Mvc

Namespace Reporting_ObjectDS_Mvc.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function Designer() As ActionResult
			Dim model As New Models.ReportDesignerModel()
			Dim dataSource As New ObjectDataSource()
			dataSource.Name = "EmployeeObjectDS"
			dataSource.DataSource = GetType(Employees.EmployeeList)
			dataSource.Constructor = ObjectConstructorInfo.Default
			dataSource.DataMember = "Items"
			dataSource.RebuildResultSchema()
			model.DataSources = New Dictionary(Of String, Object)()
			model.DataSources.Add("EmployeeObjectDS", dataSource)
			Return View(model)
		End Function

		Public Function Viewer() As ActionResult
			Return View()
		End Function
	End Class
End Namespace
